src/pymanifold.py

```diff
@@ -125,6 +125,54 @@ def channel(self,
             self.dg.edges[port_from, port_to][key] = attr
         return
 
+    def retrieve(port_in, port_out, attr):
+      if(port_out == "None"):
+        return self.dg.edges[port_in][attr]
```

Comment:
This is the right idea, however dg.edges requires that the name (port_in in this case) be a tuple in this form: ```(port_in, port_out)``` as that is the unique name for the edge (eg. for the single_channel_test the name of that edge would be ('in', 'out')). 
This line would require that port_in be a tuple containing 2 strings, so there should be an if condition ```if isinstance(port_in, list):``` nested within the above if statement and if this returns false it means that only one node was provided. In that case this function can be extended so  that it looks under dg.nodes and can be used by your later get_node_X methods. Also Python doesn't require round brackets around the condition of if statements unlike Javascript :smiley: